papermill/s3.py

```diff
@@ -88,57 +134,91 @@ def wrapper(*args, **kwargs):
                 except Exception as e:
                     logger.debug('retrying: {}'.format(e))
             else:
-                raise
-
+                raise Exception  # TODO verify what to raise
```

Comment:
Should probably save `e` and raise `e` if not None and `Exception("Unknown Failures")` otherwise?